scripts/check_gcov_coverage.py

```diff
@@ -0,0 +1,245 @@
+        changed_lines = {}
+        # Get changes for each commit in the PR range
+        commits = run_git_command(["rev-list", f"{base}...{head}"], cwd=root).split()
+        print(f"Found {len(commits)} commits in PR range:")
+        for c in commits:
+            print(f"  {c}")
+            cl = get_changed_lines(c, root)
+            for f, lines in cl.items():
+                if f not in changed_lines:
+                    changed_lines[f] = set()
+                changed_lines[f].update(lines)
+    else:
+        # Handle single commit
+        print(f"Analyzing commit '{commit}'...")
+        changed_lines = get_changed_lines(commit, root)
+    print(f"Found changes in {len(changed_lines)} files (before filtering):")
+    for filename, lines in sorted(changed_lines.items()):
+        print(f"  {filename}: {len(lines)} lines changed")
+
+    print("Getting coverage data...")
+    coverage_data, filter_pattern = get_coverage_data(root)
+
+    # Extract the regex pattern from the filter (remove root path)
+    pattern_str = filter_pattern.replace(root, "").lstrip(os.path.sep)
+    pattern = re.compile(pattern_str)
+
+    filtered_lines = {
+        f: lines for f, lines in changed_lines.items() if pattern.search(f)
+    }
+
+    print(f"After filtering, {len(filtered_lines)} files match pattern '{pattern_str}'")
+
+    total_new_lines = 0
+    covered_lines = 0
+    uncovered_lines: List[Tuple[str, int]] = []
+
+    for filename, line_numbers in filtered_lines.items():
+        file_coverage = coverage_data.get(filename, {})
+
+        for lineno in line_numbers:
+            total_new_lines += 1
+
+            if lineno in file_coverage and file_coverage[lineno] > 0:
+                covered_lines += 1
+            else:
+                uncovered_lines.append((filename, lineno))
+
+    return covered_lines, total_new_lines, uncovered_lines
+
+
+def main() -> int:
+    """Main entry point"""
+    parser = create_argument_parser()
+    args = parser.parse_args()
+
+    try:
+        script_dir = os.path.dirname(os.path.abspath(__file__))
+        root = os.path.dirname(script_dir)
+        os.chdir(root)
+        print(f"Current working directory: {root}")
+
+        covered, total, uncovered = check_commit_coverage(args.commit, root)
+        title = f" Coverage analysis results for {args.commit} "
+
+        # Print results with better formatting
+        title = f" Coverage analysis results for commit {args.commit} "
+        separator = "=" * len(title)
+        print(f"\n{separator}\n{title}\n{separator}")
+        print(f"New lines of code: {total}")
+        print(f"Covered lines: {covered}")
+        print(f"Uncovered lines: {len(uncovered)}")
+
+        if total > 0:
+            coverage_percent = (covered / total) * 100
+            print(f"Coverage: {coverage_percent:.2f}%")
+
+            # Check if coverage meets minimum requirement
+            if args.fail_under is not None and coverage_percent < args.fail_under:
```

Comment:
To logically exclude case of 0;

```suggestion
            if args.fail_under and coverage_percent < args.fail_under:
```